services/npm/npm-rating.service.js

```diff
@@ -0,0 +1,82 @@
+'use strict'
+
+const Joi = require('joi')
+const { starRating } = require('../text-formatters')
+const { BaseJsonService } = require('..')
+
+// https://api-docs.npms.io/#api-Package-GetPackageInfo
+const responseSchema = Joi.object({
+  score: Joi.object({
+    final: Joi.number().required(),
+    detail: Joi.object({
+      maintenance: Joi.number().required(),
+      popularity: Joi.number().required(),
+      quality: Joi.number().required(),
+    }),
+  }),
+}).required()
+
+module.exports = class NpmRating extends BaseJsonService {
+  static category = 'rating'
+
+  static route = {
+    base: 'npm',
+    pattern:
+      'rating/:type(maintenance|popularity|quality)?/:scope(@.+)?/:packageName',
```

Comment:
Few things on the route:

* We don't need and shouldn't have the `rating` prefix. I imagine this was added given the bundling under `npm`, but since that's changing the base will be something like `npms` or `npmsio` and the extra route segment becomes superfluous. 
* Instead of making the type/kind/variant an optional param and assuming final in the absence, we should instead simply include `final` in the allowed.

These changes ensure this route will follow our [route design guidelines](https://github.com/badges/shields/blob/master/doc/badge-urls.md)

Reply:
- same to the reply above, npms vs npm
- will update to `rating/final`

Reply:
Please take a look at the guidelines I linked in my previous response if you haven't already. `rating` should be dropped, both because these aren't really rating badges and because the noun portion will be `maintenance`, `final`, etc.

Reply:
certainly, read.

As discussed recently above, will move to `npms`, so I will drop rating.